File: hangover.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main(){
  double x;
  while(cin>>x){
    if(x==0.00){
      break;
    }
    else{
      int i=2.00;
      int count=0;
      double sum=0.00;
      while(1){
        if(sum>=x){
          break;
        }
        double temp;
        temp=1/i;
        sum+=temp;
        count++;
        i++;
        cout<<temp<<endl;
      }
      cout<<count<<" card(s)"<<endl;
    }
  }
}

File: hangover.py
while(1):
    n=input()
    if(n==0.00):
        break
    else:
        sum=0
        count=0
        i=2.00;
        while(1):
            if(sum>=n):
                break
            else:
                sum+=1/i
                i+=1
                count+=1
                #print 1/i
        print count,"card(s)"
